### PR TITLE
Create form Subject

### PR DESCRIPTION
**Pull request by limkimhuor**

### CODE REVIEW

File: app/controllers/supervisor/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Supervisor::SubjectsController < ApplicationController
+  def new
+  	@subject = Subject.new
+  end
+
+  def create
+  	@subject = Subject.new(subject_params)
```
Comment (RathanakSreang) [inline app/controllers/supervisor/subjects_controller.rb]: Remove ( )

File: app/controllers/supervisor/subjects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  		render "new"
+  	end  	
+  end
+
+  def show
+  end
+
+  private
+  def subject_params
+  	params.require(:subject).permit(:name, :description, :status)
```
Comment (RathanakSreang) [inline app/controllers/supervisor/subjects_controller.rb]: Remove ( ) from permit -> .permit :name, :description, :status

File: app/views/supervisor/subjects/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<% provide(:title, 'Create subject') %>
```
Comment (RathanakSreang) [inline app/views/supervisor/subjects/new.html.erb]: User " instead of ' and remove ()

File: app/views/supervisor/subjects/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<% provide :title, "Create new subject" %>
+<h1>Create new subject</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for [:supervisor, @subject] do |f| %>
+      <%= render 'shared/error_messages' %>
```
Comment (RathanakSreang) [inline app/views/supervisor/subjects/new.html.erb]: Use `"` instead of `'`

File: app/views/supervisor/subjects/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+      <div class="field">
+        <div>
+          <%= f.hidden_field :status, :value => "start" %>
```
Comment (RathanakSreang) [inline app/views/supervisor/subjects/new.html.erb]: ` value: "start"

File: app/views/supervisor/subjects/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        </div>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: 'form-control' %>
+
```
Comment (RathanakSreang) [inline app/views/supervisor/subjects/new.html.erb]: Use `"` instead of `'`

File: app/views/supervisor/subjects/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      <%= f.label :description %>
+      <%= f.text_area :description, class: 'form-control' %>
+      </div>
```
Comment (RathanakSreang) [inline app/views/supervisor/subjects/new.html.erb]: Use `"` instead of `'`

File: app/views/shared/_error_messages.html.erb [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
     </div>
     <ul>
-    <% object.errors.full_messages.each do |msg| %>
+    <% @subject.errors.full_messages.each do |msg| %>
       <li><%= msg %></li>
     <% end %>
     </ul>
   </div>
-<% end %>
+<% end %>
```
Comment (RathanakSreang) [inline app/views/shared/_error_messages.html.erb]: Add new line

File: app/views/shared/_error_messages.html.erb [hunk 9]
```diff
@@ -1,12 +1,12 @@
-<% if object.errors.any? %>
+<% if @subject.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      The form contains <%= pluralize object.errors.count, "error" %>
+      The form contains <%= pluralize(@subject.errors.count, "error") %>.
```
Comment (RathanakSreang) [inline app/views/shared/_error_messages.html.erb]: remove ( )

**Comment by phornsopheak** [comment]
it's ok.

**Comment by samnangcattor** [comment]
ok

**Comment by RathanakSreang** [comment]
Ok

File: Gemfile [hunk 10]
```diff
@@ -6,6 +6,7 @@ gem "bcrypt", "3.1.7"
 gem "bootstrap-sass", "3.2.0.0"
 gem "will_paginate", "3.0.7"
 gem "bootstrap-will_paginate", "0.0.10"
+gem "nested_form"
```
Comment (dieunb) [inline Gemfile]: remove

**Comment by phornsopheak** [comment]
ok

**Comment by samnangcattor** [comment]
ok

File: app/controllers/supervisor/tasks_controller.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+class Supervisor::TasksController < ApplicationController
```
Comment (dieunb) [inline app/controllers/supervisor/tasks_controller.rb]: remove this file, don't need

File: config/routes.rb [hunk 12]
```diff
@@ -7,5 +7,7 @@
   namespace :supervisor do
     resources :users
     resources :courses
+    resources :subjects
+    resources :tasks
```
Comment (dieunb) [inline config/routes.rb]: remove

**Pull request merged**